qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/common/data_handling/data_model.py

```diff
@@ -484,7 +398,7 @@ def get_all_workspace_names(self):
         return self._data_workspaces.get_loaded_workpace_names() + self._data_workspaces.get_bgsub_workpace_names()
 
     def get_log_workspaces_name(self):
-        return [ws.name() for ws in self._log_workspaces] if self._log_workspaces else ""
+        return self._log_workspaces.name()
```

Comment:
If I load a focused data file on the Fitting tab an then click the "Remove All" button I get an error from this line due to the fact that self._log_workspaces is None (`AttributeError: 'NoneType' object has no attribute 'name'`). Could the "protection" given by the `if self._log_workspaces else ""` check be reinstated here?

Also I'm wondering whether the behaviour where this function previously returned a list should be reinstated. Perhaps we need two functions here:
- `get_log_workspaces_names` (renamed version of original) which returns the individual log workspace names that are in the group eg run_info etc. The function is called from remove_workspace in data_presenter in a way that expects a list. To avoid the rather nasty feature in the original code where it had a hybrid return type it should perhaps do:
```
return [ws.name() for ws in self._log_workspaces] if self._log_workspaces else [""]
```
- `get_log_workspaces_name` (new function) which returns the name of the log workspaces group. I see you use this in the fit_completed code